### PR TITLE
Prepare Release 2.5.1

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Prepare Release 2.5.1

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-k8s-operator/issues/709 and https://github.com/opensearch-project/opensearch-k8s-operator/issues/674

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by bbarani — APPROVED** [review]

File: charts/opensearch-operator/values.yaml [hunk 1]
```diff
@@ -41,7 +41,7 @@ manager:
   parallelRecoveryEnabled: true
 
   image:
-    repository: public.ecr.aws/opensearch-project/opensearch-operator
+    repository: opensearchproject/opensearch-operator
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply charts/opensearch-operator/values.yaml:44]: With this change the project is consistent with DockerHub and also fixing the wrong image information.

File: jenkins/release.jenkinsfile [hunk 2]
```diff
@@ -120,7 +120,8 @@ pipeline {
                                 string(name: 'RELEASE_VERSION', value: "${OPERATOR_VERSION}"),
                                 booleanParam(name: 'DOCKER_HUB_PROMOTE', value: true),
                                 booleanParam(name: 'ECR_PROMOTE', value: true),
-                                booleanParam(name: 'TAG_LATEST', value: true)
+                                booleanParam(name: 'TAG_LATEST', value: true),
+                                booleanParam(name: 'TAG_MAJOR_VERSION', value: false)
```
Comment (gaiksaya) [inline jenkins/release.jenkinsfile:124]: Don't we want to update the major version tag? Example: 2.5.1 -> 2

File: charts/opensearch-operator/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaiksaya) [reply charts/opensearch-operator/values.yaml:44]: Thanks!

**Comment by vipinjn24** [comment]
Please verify the image is accessible
image

Also please update the opensearch versions we can install with this operator

File: jenkins/release.jenkinsfile (hunk 2 above)
Comment (prudhvigodithi) [reply jenkins/release.jenkinsfile:124]: Adding @swoehrl-mw.
Sayali mostly the operator is installed and managed via helm and helm charts have fixed versions in `Chart.yml`, `docker run` wont work here hence I dont see a need to keep `TAG_MAJOR_VERSION` which adds more noise.

**Comment by prudhvigodithi** [comment]
Hey @vipinjn24 please check the the workaround I have posted here https://github.com/opensearch-project/opensearch-k8s-operator/issues/709#issuecomment-1922632388, also to fix this I have a raised a PR for `2.5.1` release.
https://github.com/opensearch-project/opensearch-k8s-operator/pull/716

File: jenkins/release.jenkinsfile (hunk 2 above)
Comment (prudhvigodithi) [reply jenkins/release.jenkinsfile:124]: The [docker-promotion](https://build.ci.opensearch.org/job/docker-promotion/) job has default `TAG_MAJOR_VERSION` to `true` hence disabling it.

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**